File: lang/HbFalse.py
from HbObject import HbObject

class HbFalse(HbObject):
  
  classname = 'FalseClass'
  superclass = 'Object'
  
  def __init__(self):
    HbObject.__init__(self)
    self.methods.update({
      
    })
  
  def to_bool(self):
    return self
  
  def to_string(self):
    from HbString import HbString
    return HbString('false')
  

File: src/hobbes/values/HbFalse.java
package hobbes.values;

import hobbes.interpreter.Interpreter;

@HobbesClass(name="FalseClass")
public class HbFalse extends HbObject {
	
	public HbFalse(Interpreter o) {
		super(o);
	}
	
	@HobbesMethod(name="toBool")
	public HbObject toBool() {
		return this;
	}
	
	@HobbesMethod(name="show")
	public HbString hbShow() {
		return getObjSpace().getString("false");
	}
	
	@HobbesMethod(name="toString")
	public HbString hbToString() {
		return hbShow();
	}

}
